service/s3/s3manager/upload.go

```diff
@@ -508,6 +508,7 @@ func (u *multiuploader) upload(firstBuf io.ReadSeeker) (*UploadOutput, error) {
 
 	// Read and queue the rest of the parts
 	for u.geterr() == nil {
+		num++
```

Comment:
Yea, the unit test looks fine. So, with that said, your change is fine, but doesn't mean we've hit the `io.EOF` for the DL. I also believe this means we would never even hit the `io.ErrUnexpectedEOF` case either. So, I don't know if that is something we'd like to have changed. We would like to end when we know that there is no bytes left and not by the amount of parts we've iterated.